### PR TITLE
Implement public organization mediafiles

### LINKED ISSUE #2499: Allow orga-files/folders to be accessible in meetings
**Desired behaviour:**
- A Orgaadmin and a superadmin should now be allowed to set files/folders to public
- Files/folders should only be allowed to be set to public on the orga-level

### PR DESCRIPTION
**Pull request by luisa-beerboom**
Closes #2499 
Needs OpenSlides/openslides-meta#146
Edited https://github.com/OpenSlides/OpenSlides/wiki/Mediafiles

TODO:
- [x] mediafile.delete
- [x] mediafile.move
- [x] mediafile.update
- [x] meeting.clone
- [x] meeting.delete
    - [x] A test for if orga mediafiles are deleted if they have meeting data
- [x] meeting.import
- [x] meeting.set_font
- [x] meeting.set_logo
- [x] meeting.unset_font
- [x] meeting.unset_logo
- [x] motion.create (and topic and assignment)
- [x] projector.project (Needs to exchange mediafile fqids for meeting_mediafile fqids)
- [x] presenter check_database
- [x] presenter check_database_all
- [x] presenter check_mediafile_id will be deleted soon, don't bother
- [x] Test everything with implicit meeting_mediafiles (should be treated as if `inherited_access_group_ids == [admin_group_id]`)
   - [x] mediafile.move
   - [x] mediafile.update
   - [x] meeting.clone
   - [x] meeting.import
   - [x] meeting.set_font
   - [x] meeting.set_logo
   - [x] meeting.unset_font
   - [x] meeting.unset_logo
   - [x] motion.create (and topic and assignment)
   - [x] presenter check_database
   - [x] presenter check_database_all
- [x] Wiki
   - [x] delete presenter 
- [x] Implement delete presenter
- [x] See if logo setting, attachment generation and projecting falsely works with non-published orga files
- [x] Database checker should use admin_id for inherited_access_groups if there is no meeting mediafile for the parent -> Test this!

Then:
- [x] Implement set public in orga functionality
- [x] Wiki
- [x] Mediafiles entry in main OpenSlides Wiki

### CODE REVIEW

File: tests/system/action/meeting_mediafile/test_delete.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+from tests.system.action.base import BaseActionTestCase
+
+
+class MeetingMediafileDelete(BaseActionTestCase):
+    def test_delete(self) -> None:
+        self.create_meeting()
+        self.set_models(
```
Comment (r-peschke) [inline tests/system/action/meeting_mediafile/test_delete.py:7]: Please write a test with more or less all fields, especially the relations list_of_speakers, assignments , ... and show that they are handled as expected

**Review by r-peschke — CHANGES_REQUESTED** [review]
I'm still not ready reviewing (51/73), but will continue tomorrow.

File: tests/system/action/assignment/test_create.py [hunk 2]
```diff
@@ -131,20 +139,61 @@ def test_create_full_fields(self) -> None:
                 "phase": "search",
                 "default_poll_description": "text_test2",
                 "number_poll_candidates": True,
+                "attachment_ids": [1],
```
Comment (r-peschke) [inline tests/system/action/assignment/test_create.py:142]: please rename attachment_ids to mediafile_ids to reflect the collection name in the field name

File: tests/system/action/assignment/test_create.py [hunk 3]
```diff
@@ -131,20 +139,61 @@ def test_create_full_fields(self) -> None:
...
+            "assignment/1",
+            {
+                "title": "test_Xcdfgee",
+                "meeting_id": 110,
+                "description": "text_test1",
+                "open_posts": 12,
+                "phase": "search",
+                "default_poll_description": "text_test2",
+                "number_poll_candidates": True,
+                "attachment_ids": [11],
```
Comment (r-peschke) [inline tests/system/action/assignment/test_create.py:156]: Please rename this `attachment_ids` to `meeting_mediafile_ids` to reflect also the collection name

File: tests/system/migrations/test_0054_split_mediafile_model.py [hunk 4]
```diff
@@ -0,0 +1,384 @@
+def test_migration(write, finalize, assert_model):
+    write(
+        {
```
[the remark this answers is not in the document]
Comment (luisa-beerboom) [reply tests/system/migrations/test_0054_split_mediafile_model.py:3]: Not applicable in this particular place

File: openslides_backend/action/actions/mediafile/move.py [hunk 5]
```diff
@@ -47,27 +51,47 @@ def get_updated_instances(self, action_data: ActionData) -> ActionData:
         action_data = super().get_updated_instances(action_data)
         # Action data is an iterable with exactly one item
         instance = next(iter(action_data))
+        owner_collection, owner_id_str = str(instance["owner_id"]).split(KEYSEPARATOR)
```
[the remark this answers is not in the document]
Comment (luisa-beerboom) [reply openslides_backend/action/actions/mediafile/move.py:54]: Done

File: tests/system/action/meeting_mediafile/test_delete.py (hunk 1 above)
Comment (luisa-beerboom) [reply tests/system/action/meeting_mediafile/test_delete.py:7]: Done

File: tests/system/action/assignment/test_create.py (hunk 2 above)
Comment (luisa-beerboom) [reply tests/system/action/assignment/test_create.py]: Done something like that

File: tests/system/action/assignment/test_create.py (hunk 3 above)
Comment (luisa-beerboom) [reply tests/system/action/assignment/test_create.py]: Done something like that

**Comment by bastianjoel** [comment]
There is something wrong with the calculation of `inherited_access_groups`. 
I set access groups of a published file within a published folder in a meeting. The `access_group_ids` were updated but `inherited_access_group_ids` were not updated. 

Editing the parent folder also did not change the `inherited_access_group_ids` of that file.

**Review by rrenkert — APPROVED** [review]

**Pull request merged**